File: backend-spring/shopapp/src/main/java/com/ducbn/shopapp/models/Product.java
package com.ducbn.shopapp.models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "products")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Product extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 350)
    private String name;

    private  Float price;

    @Column(name = "thumbnail", length = 300)
    private  String thumbnail;

    @Column(name = "description")
    private String description;



    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: shopapp-angular/src/app/models/product.ts
import { Category } from "./category";
import { ProductImage } from "./product.image";

export interface Product {
  id: number;
  name: string;
  price: number;
  thumbnail: string;
  description: string;
  category: number;
  url: string;
  product_images: ProductImage[];
}
